clients/drcachesim/common/trace_entry.h

```diff
@@ -191,6 +191,35 @@ typedef enum {
      */
     TRACE_MARKER_TYPE_CPU_ID,
 
+    /**
+     * The marker value contains function id defined by the user in droption.
+     */
+    TRACE_MARKER_TYPE_FUNC_ID,
+
+    // XXX i#3048: replace return address with callstack information.
+    /**
+     * The marker value contains the return address of the function call,
+     * whose id is specified by the closest previous
+     * marker entry.
+     */
+    TRACE_MARKER_TYPE_FUNC_RETADDR,
+
+    /**
+     * The marker value contains one argument value of the function call,
+     * whose id is specified by the closest previous #TRACE_MARKER_TYPE_FUNC_ID
+     * marker entry. The number of such entries for one function invocation is
+     * expected to be equal to the specified arugment in -record_function or
```

Comment:
spelling (please write these user-facing docs carefully, just like you would with code): argument

Reply:
Done